ZeroWasteFriendly/Repository/PlacesRepository.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  PlacesRepository.swift
+//  ZeroWasteFriendly
+//
+//  Created by Marek Skrzelowski on 21/02/2020.
+//  Copyright © 2020 vandermesis. All rights reserved.
+//
+
+import Foundation
+import CloudKit
+
+typealias FetchPlacesCompletion = (Result<[Place], Error>) -> Void
+
+protocol PlacesRepository {
+    func fetchPlaces(completion: FetchPlacesCompletion?)
+}
+
+final class PlacesRepositoryImpl: BaseCloudKitService, PlacesRepository {
+
+    func fetchPlaces(completion: FetchPlacesCompletion?) {
+        fetchPublicCloudKitDB(recordType: .recordEcoPlaces) { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case .success(let records):
+                let places = self.convertRecordsToPlaces(records: records)
+                completion?(.success(places))
+            case .failure(let error):
+                completion?(.failure(error))
+            }
+        }
+    }
+}
+
+private extension PlacesRepositoryImpl {
+
+    private func convertRecordsToPlaces(records: [CKRecord]) -> [Place] {
```

Comment:
I've never tried this, but maybe it is possible to use Codable Protocol and do this automaticaly?

Reply:
Unfortunatelly there is no way to use Codable to convert CloudKit records. I found only two third party libraries on github. And they are not well tested.